src/backend/looped/weapons/paintgun.cpp

```diff
@@ -0,0 +1,50 @@
+#include "backend/looped/looped.hpp"
+#include "core/enums.hpp"
+#include "gta/enums.hpp"
+#include "util/entity.hpp"
+#include "util/color.hpp"
+
+namespace big
+{
+	void looped::weapons_paint_gun()
+	{
+		if (g.weapons.paintgun.rainbow)
+		{
+			g.weapons.paintgun.col[0] = rgb.red;
+			g.weapons.paintgun.col[1] = rgb.green;
+			g.weapons.paintgun.col[2] = rgb.blue;
+		}
+
+		if (g.weapons.custom_weapon == CustomWeapon::PAINT_GUN && (!g.self.custom_weapon_stop || WEAPON::IS_PED_ARMED(self::ped, 4 | 2)))
+		{
+			Vector3 c; entity::raycast(&c);
+
+			if (PAD::IS_DISABLED_CONTROL_PRESSED(0, (int)ControllerInputs::INPUT_AIM))
+			{
+				if (PAD::IS_DISABLED_CONTROL_PRESSED(0, (int)ControllerInputs::INPUT_ATTACK))
+				{
+					GRAPHICS::ADD_DECAL((int)DecalTypes::splatters_paint,
+					    c.x,
+					    c.y,
+					    c.z,
+					    0,  //true
+					    0,  //true
+					    -1, //true
+					    0,
+					    1.f,
+					    0.f, // always 0
+					    0.5f, //size x
+					    0.4f, //size y
+					    g.weapons.paintgun.col[0],
+					    g.weapons.paintgun.col[1],
+					    g.weapons.paintgun.col[2],
+					    g.weapons.paintgun.col[3],
+					    -1,
+					    true,
+					    false,
+					    false);
+				}
+			}
+		}
+	}
+}
```

Comment:
This can be made into a looped command.

Reply:
Can it?
It is using the custom weapon combo and no other members of it use commands.
https://github.com/YimMenu/YimMenu/blob/61bb60d1f430c932b3870478ba093d96d2895b6f/src/views/self/view_weapons.cpp#L107-L128

Reply:
Probably not, I was a bit trigger happy with saying to use the looped command stuff.

For this specific example there's currently no way to implement it.

Reply:
resolving this and tp gun as of comment above. Unless it is wanted a checkbox toggle which would be unnecessary.